props/saber_sa22c_buttons.h

```diff
@@ -0,0 +1,334 @@
+// sa22c props file, includes the following changes
+//
+// New #define NO_LOCKUP_HOLD
+// reverts to lockup being triggered only by clash + pow or aux
+// Button configs:
+//
+// 1 Button:
+// Activate Muted - fast double click while OFF
+// Activate - short click while OFF
+// Play/Stop Music - hold and release while ON
+// Turn off blade - hold and wait till blade is off while ON
+// Next Preset - hold and release while OFF
+// Prev Preset - hold and wait while OFF
+// Lockup - hold + hit clash while ON
+// Drag - hold + hit clash while ON pointing the blade tip down
+// Blaster Blocks - short click while ON
+// Force Effects - hold + twist the hilt while ON (while pointing up)
+// Color Change mode - hold + twist the hilt while ON (pointing down)
+//
+// 2 Button:
+// POWER
+// Activate Muted - fast double click while OFF
+// Activate - short click while OFF
+// Play/Stop Music - hold and release while OFF
+// Turn off blade - hold and wait till blade is off while ON
+// Force Effects - double click while ON
+// Volume UP - short click while OFF and in VOLUME MENU
+// Prev Preset - hold and wait while OFF
+// Color Change mode - hold + toggle AUX while ON
+// AUX
+// Blaster blocks - short click while ON
+// Next Preset - short click while OFF
+// Lockup - hold while ON
+// Drag - hold while ON pointing the blade tip down
+// Enter VOLUME MENU - long click while OFF
+// Volume down - short click while OFF and in VOLUME MENU
+// Battery level - hold while off
+//
+// 3 Button: Same as two button except for VOLUME MENU
+// AUX
+// Volume UP - short click while OFF and in VOLUME MENU
+// AUX2
+// Lockup - hold while ON
+// Previous Preset - short click while OFF
+// Volume DOWN - short click while OFF and in VOLUME MENU
+
+#ifndef PROPS_SABER_SA22C_BUTTONS_H
+#define PROPS_SABER_SA22C_BUTTONS_H
+
+#include "prop_base.h"
+
+#undef PROP_TYPE
+#define PROP_TYPE SaberSA22CButtons
+
+// The Saber class implements the basic states and actions
+// for the saber.
+class SaberSA22CButtons : public PropBase {
+public:
+  SaberSA22CButtons() : PropBase() {}
+  const char* name() override { return "SaberSA22CButtons"; }
+
+  void ChangeVolume(bool up) {
+    if (up) {
+      STDOUT.println("Volume up");
+      if (dynamic_mixer.get_volume() < VOLUME) {
+        dynamic_mixer.set_volume(dynamic_mixer.get_volume() + VOLUME * 0.10);
+        beeper.Beep(0.5, 2000);
+        STDOUT.print("Current Volume: ");
+        STDOUT.println(dynamic_mixer.get_volume());
+      }
+      else {
+        beeper.Beep(0.5, 3000);
+      }
+    } else {
+      STDOUT.println("Volume Down");
+      if (dynamic_mixer.get_volume() > (0.10 * VOLUME)) {
```

Comment:
Do you want the minimum volume to be zero or 10% of VOLUME?
Currently this allows the volume to go near zero, but not all the way to zero, which is kind of odd.
It might make more sense to do something like:

if (dynamic_mixer.get_volume() > 0) {
   dynamic_mixer.set_volume(std::min&lt;int&gt;(dynamic_mixer.get_volume() - VOLUME * 0.1));

Reply:
I had wanted it not to totally mute because then it's hard to get in and out of the volume menu and navigate the saber, etc.  So its intended to allow volume to range from 100 - 10 %